packages/api/src/routes/internal/hie/commonwell.ts

```diff
@@ -107,4 +111,43 @@ router.put(
   })
 );
 
+/**
+ * POST /internal/commonwell/patient-discovery/:patientId
+ *
+ * Triggers patient discovery for a specific patient in CommonWell.
+ * @param req.query.cxId The customer ID.
+ * @param req.params.patientId The ID of the patient to run discovery for
+ * @param req.query.rerunPdOnNewDemographics Optional flag to rerun discovery if new demographics are found
+ * @param req.query.forceCommonwell Optional flag to force CommonWell discovery
+ * @returns 200 OK if discovery was triggered successfully
+ */
+router.post(
+  "/patient-discovery/:patientId",
+  handleParams,
+  requestLogger,
+  asyncHandler(async (req: Request, res: Response) => {
+    const cxId = getUUIDFrom("query", req, "cxId").orFail();
+    const patientId = getFrom("params").orFail("patientId", req);
+    const rerunPdOnNewDemographics = stringToBoolean(
+      getFrom("query").optional("rerunPdOnNewDemographics", req)
+    );
+    const forceCommonwell = stringToBoolean(getFrom("query").optional("forceCommonwell", req));
```

Comment:
we have a `getFromQueryAsBoolean` util in api/src/routes/util.ts, lets use that!